addons/analytic/models/analytic_line.py

```diff
@@ -198,3 +198,18 @@ class AccountAnalyticLine(models.Model):
         [('other', 'Other')],
         default='other',
     )
+
+    def write(self, vals):
+        res = super().write(vals)
+        if 'amount' in vals or 'account_id' in vals:
+            for line in self:
+                move_line = self.env['account.move.line'].browse(line.move_line_id)
+                move_line._compute_analytic_distribution()
```

Comment:
This will get the distribution based on the `account.analytic.distribution.model` setup, not based on the changes done in the analytic lines